### PR TITLE
Border.Shadow hide the collectionView Header/footer -UITest

### LINKED ISSUE #19338: [iOS] Border.Shadow hide the collectionView Header
### Description

I have created a collectionView with a header, which contains a border.

If I apply shadow style to the border, the header disappears.

**Expectation**
image

**Reality**
image

### Steps to Reproduce

1. Create a new MAUI project.
2. Create a CollectionView with a header.
3. Set a border component in the header.
4. Apply shadow style to the border.
5. Launch the application and check that the header is not displayed.

### Link to public reproduction project repository

[Link to the repository](https://github.com/MADSENSE/MADSENSE.MAUI.Sample.App/tree/border-shadow-collectionView-header)

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.3

### Did you find any workaround?

If you set the border component inside a ContentView, works fine for me.

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

Whenever a view, wrapped inside a wrapper view, has been a direct child of a header or footer element, it either didn't show at all or rendered incorrectly.

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/19338

|Before|After|
|--|--|
|||
|||

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue19338.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		}
+
+		public override string Issue => "Border.Shadow hide the collectionView Header";
+
+		[Test]
+		[Category(UITestCategories.CollectionView)]
+		public void Issue935TestsMultipleOnTappedViewCell()
+		{
+			App.WaitForElement("HeaderLabel");
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue19338.cs:20]: Snapshots available in the latest build:
image
Could you commit the images?

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue19338.cs (hunk 1 above)
Comment (kubaflo) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue19338.cs:20]: I had to add margin to the collection, so I need to re-upload screenshots
HeaderAndFooterShouldBeVisible

Comment (jsuarezruiz) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue19338.cs:20]: The test **HeaderAndFooterShouldBeVisible** is failing on all the platforms. After the margin change, need to update the snapshot in all the platforms. Images available in the latest build.

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]
The test **HeaderAndFooterShouldBeVisible** is failing on all the platforms. After the margin change, need to update the snapshot in all the platforms. Images available in the latest build.

**Review by rmarinho — COMMENTED** [review]
Does this work fine with CollectionView2 ?

**Comment by kubaflo** [comment]
Hi, @rmarinho I was testing it on the sandbox project on the latest main which I think uses cv2, right?

**Comment by rmarinho** [comment]
Seems the test is failing here

 https://dev.azure.com/xamarin/public/_build/results?buildId=129415&view=ms.vss-test-web.build-test-results-tab&runId=3432248&resultId=100034&paneView=attachments
 
 
image

**Review by rmarinho — CHANGES_REQUESTED** [review]
https://dev.azure.com/xamarin/public/_build/results?buildId=129494&view=ms.vss-test-web.build-test-results-tab&runId=3436171&resultId=100052&paneView=attachments

image

**Comment by kubaflo** [comment]
@rmarinho do UI tests use CV1 or CV2 or both? I noticed that the shadow of the header of CV1 (right screenshot) is slightly different than the one of CV2 (left screenshot)
Screenshot 2024-12-04 at 12 20 56

**Comment by rmarinho** [comment]
> @rmarinho do UI tests use CV1 or CV2 or both? I noticed that the shadow of the header of CV1 (right screenshot) is slightly different than the one of CV2 (left screenshot) Screenshot 2024-12-04 at 12 20 56

the CI runs with both , seems is failing for CV1 and passing for CV2

**Comment by kubaflo** [comment]
> > @rmarinho do UI tests use CV1 or CV2 or both? I noticed that the shadow of the header of CV1 (right screenshot) is slightly different than the one of CV2 (left screenshot) Screenshot 2024-12-04 at 12 20 56
> 
> the CI runs with both , seems is failing for CV1 and passing for CV2

I see, so basically currently the tests will never pass. Therefore, I've created this PR: https://github.com/dotnet/maui/pull/26360 what do you think?

The other solution would be to enable clipping for cv1

**Comment by kubaflo** [comment]
Rebased, so now Ui tests should pass

File: src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs [hunk 2]
```diff
@@ -167,6 +167,11 @@ internal void UpdateSubview(object view, DataTemplate viewTemplate, nint viewTag
 
 			if (uiView != null)
 			{
+				if (uiView.Superview is WrapperView wrapperView)
```
Comment (PureWeen) [inline src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs:170]: I'm wondering if the uiView passed into this view is the wrong view

Line `158` calls `RemovefromSuperView`

If uIVIew has a `WrapperView` that's going to cause some weird behavior

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs (hunk 2 above)
Comment (kubaflo) [reply src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs:170]: The uiView passed to this view is a Wrapper view, but when it goes here: UpdateView(view, viewTemplate, ref uiView, ref formsElement); Line `166` the wrapper view is replaced by the view it wraps

Comment (kubaflo) [reply src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs:170]: Here: https://github.dev/kubaflo/maui/blob/0975e08cc5a3c78b30918e39cbe532d9df4ed173/src/Controls/src/Core/Handlers/Items/iOS/TemplateHelpers.cs#L29

Comment (PureWeen) [reply src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs:170]: I wonder if the overall fix here would be to return "ToPlatform()" from `RealizeView` vs ".PlatformView"

Comment (kubaflo) [reply src/Controls/src/Core/Handlers/Items/iOS/StructuredItemsViewController.cs]: It works!!

**Review by rmarinho — CHANGES_REQUESTED** [review]
Can we rebase and see how it works with the new container?

**Comment by kubaflo** [comment]
@rmarinho yess it works with the new generalWrapperView! But, maybe let's keep the UI test?

**Review by rmarinho — APPROVED** [review]

**Review by rmarinho — CHANGES_REQUESTED** [review]
Do we need new images ? 

CV1
image

CV2
image

**Comment by rmarinho** [comment]
Diff 
image

**Comment by kubaflo** [comment]
@rmarinho I think the tests passed for this issue

**Review by rmarinho — APPROVED** [review]

**Pull request merged**